src/Sylius/Bundle/LocaleBundle/Checker/LocaleUsageChecker.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\LocaleBundle\Checker;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Sylius\Component\Locale\Context\LocaleNotFoundException;
+use Sylius\Component\Resource\Metadata\RegistryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+final class LocaleUsageChecker implements LocaleUsageCheckerInterface
+{
+    public function __construct(
+        private RepositoryInterface $localeRepository,
+        private RegistryInterface $resourceRegistry,
+        private EntityManagerInterface $entityManager,
+    ) {
+    }
+
+    /**
+     * @throws LocaleNotFoundException
+     */
+    public function isUsed(string $localeCode): bool
+    {
+        $locale = $this->localeRepository->findOneBy(['code' => $localeCode]);
+
+        if (null === $locale) {
+            throw new LocaleNotFoundException();
+        }
+
+        $translationEntityInterfaces = $this->getTranslationEntityInterfaces();
+
+        foreach ($translationEntityInterfaces as $entityInterface) {
+            if ($this->isLocaleUsedByTranslation($entityInterface, $localeCode)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @return array<int, class-string>
+     */
+    private function getTranslationEntityInterfaces(): array
+    {
+        $translationEntityInterfaces = [];
+
+        foreach ($this->resourceRegistry->getAll() as $resource) {
+            $resourceParameters = $resource->getParameters();
+
+            if (isset($resourceParameters['translation']['classes']['interface'])) {
+                $translationEntityInterfaces[] = $resourceParameters['translation']['classes']['interface'];
+            }
+        }
+
+        return $translationEntityInterfaces;
+    }
+
+    private function isLocaleUsedByTranslation(string $translationEntityInterface, string $localeCode): bool
+    {
+        $repository = $this->entityManager->getRepository($translationEntityInterface);
+
+        return $repository->count(['locale' => $localeCode]) > 0;
```

Comment:
That's a lot of additional queries for simple delete. We would request up to n queries equal to the number of translatable entities in the system. Can't we base it on `ForeignKeyConstraintViolationException` or something similar?

Reply:
There's no foreign key to `sylius_locale`. 
<img width="482" alt="CleanShot 2023-06-09 at 07 40 46@2x" src="https://github.com/Sylius/Sylius/assets/80641364/1e9f1ce8-ad63-47e4-b7ea-cca60c26cfa6">

I know it might seem like many queries, but:
- Removing a locale is not a standard action for e-commerce (usually, store administrators don't delete a locale every morning)
- This operation might be performed by a small group of people
- In the best scenario it'd be only 1 query, at worst a number of `*Translation` entities.

I believe it won't affect any store.